azurelinuxagent/common/protocol/wire.py

```diff
@@ -1024,18 +1031,45 @@ def send_request_using_appropriate_channel(self, direct_func, host_func):
         try:
             ret = host_func()
         except (ResourceGoneError, InvalidContainerError) as e:
+            old_container_id = self.host_plugin.container_id
+            old_role_config_name = self.host_plugin.role_config_name
+
             msg = "Request failed with the current host plugin configuration." \
                   "ContainerId: {0}, role config file: {1}. Fetching new goal state and retrying the call." \
-                  "Error: {2}".format(self.host_plugin.container_id, self.host_plugin.role_config_name, ustr(e))
+                  "Error: {2}".format(old_container_id, old_role_config_name, ustr(e))
             logger.info(msg)
 
             self.update_goal_state(forced=True)
+
+            new_container_id = self.host_plugin.container_id
+            new_role_config_name = self.host_plugin.role_config_name
             msg = "Host plugin reconfigured with new parameters. " \
-                  "ContainerId: {0}, role config file: {1}.".format(self.host_plugin.container_id,
-                                                                    self.host_plugin.role_config_name)
+                  "ContainerId: {0}, role config file: {1}.".format(new_container_id, new_role_config_name)
             logger.info(msg)
 
-            ret = host_func()
+            try:
+                ret = host_func()
+                if ret:
+                    msg = "Request succeeded using the host plugin channel after goal state refresh." \
+                          "ContainerId changed from {0} to {1}, " \
+                          "role config file changed from {2} to {3}.".format(old_container_id, new_container_id,
+                                                                             old_role_config_name, new_role_config_name)
+                    add_event(name=AGENT_NAME,
+                              version=CURRENT_VERSION,
+                              op=WALAEventOperation.HostPlugin,
+                              is_success=True,
+                              message=msg,
+                              log_event=False)
+
+            except Exception as e:
+                add_event(name=AGENT_NAME,
```

Comment:
why restrict the error to only ResourceGone and InvalidContainer? Shouldn't we log that we failed irrespective of the error type?

Reply:
the intention is to measure whether the refresh helped to solve ResourceGone/InvalidContainer
if retry fails with ResourceGone/InvalidContainer then it didn't help; if it fails with some other error then it's something else

Reply:
> if it fails with some other error then it's something else
you may want to measure whether the retry introduced another issue, but i wouldn't go there

Reply:
I agree to the measurement but we should filter out ResourceGone/InvalidContainer in the query rather than out here in the source. Logging all error might help us investigate unknown issues later on

Reply:
if you want to measure failures communicating with the hostgaplugin then we need more instrumentation (e.g. failures in the first call, not only on the retry)

Reply:
btw, I am thinking these measurements should be only temporary --- we are measuring the effectiveness of specific fixes, and we may end up accummulating tons of those over time
we should have an offline talk about this

Reply:
Agree with Norberto. Since the purpose of this event is to merely gather data about the fix for `ResourceGoneError` and `InvalidContainerError`, it makes sense to only emit events for those exceptions.